transcom/mymove
# MB-7682 TOO can view cancelled shipment

## Description

This PR enables the TOO to be able to see the cancelled shipments. TOO will request the shipment to be cancelled and then the Prime will follow up by pushing the shipment status to be cancelled.

## Reviewer Notes

- Took the liberty of refactoring the devseed script a bit. There's a new flag `namedSubScenario` that will run a specific scenario within the main script file. This will shorten the time the db will take to seed data.
- Added new make commands that will backup, restore and cleanup the dev database.
- Noticed that the status name is `CANCELED` instead of `CANCELLED` which we should fix later. I'll make a new story to capture this.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

Reset and migrate the database. Delete bin file and build the generate-test-data tool.
```sh
make db_dev_reset db_dev_migrate

rm -rf bin/generate-test-data
make bin/generate-test-data
```

Generate seed data shipment hhg cancelled.
```sh
generate-test-data --named-scenario="dev_seed" --named-sub-scenario="shipment_hhg_cancelled" --db-env="development"
```

Run app
```sh
make server_run
make office_client_run
```

### Test
1. In office app, create a new TOO user to log in
2. Click on generated move in queue
3. On move details page, see that the shipment card shows the `CANCELLED` tag
4. Navigate to move task order page by clicking on the `Move task order` link at the top
5. On page, notice that `CANCELLED` tag shows next to the header and no action buttons are showing

### Storybook

```sh
make storybook
```

1. See `CANCELLED` tag on Shipment display component: http://localhost:6006/?path=/story/office-components-shipment-display--cancelled-shipment
2. See `CANCELLED` tag on Shipment heading component: http://localhost:6006/?path=/story/office-components-shipmentheading--shipment-heading-cancelled

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7682) for this change

## Screenshots

Move details page
![image](https://user-images.githubusercontent.com/1522549/122295925-a30b7300-ceae-11eb-8c38-a9dd4af81e4d.png)


Move task order page
![image](https://user-images.githubusercontent.com/1522549/122295857-92f39380-ceae-11eb-980c-d73f983ad37e.png)
